src/disk_io_thread.cpp

```diff
@@ -270,7 +270,8 @@ constexpr disk_job_flags_t disk_interface::cache_hit;
 		// see also the comment in thread_fun
 		std::unique_lock<std::mutex> l(m_job_mutex);
 		if (m_abort.exchange(true)) return;
-		bool const no_threads = m_num_running_threads == 0;
+
+		bool const no_threads = num_threads() == 0;
```

Comment:
`m_num_running_threads` may be zero here because none of the threads have started yet. Instead, look at the number of threads we're *configured* to have.